python/lib/core/dmod/core/common/helper_functions.py

```diff
@@ -27,31 +32,31 @@ def get_mro_names(value) -> list[str]:
 
 
 def is_integer(value) -> bool:
-    if type(value) != type:
-        value = value.__class__
-
-    mro = get_mro_names(value)
+    if numpy and isinstance(value, numpy.integer):
+        return True
 
-    module_name = value.__module__ if hasattr(value, "__module__") else ''
+    loaded_integer_types = [int] + get_subclasses(int)
 
-    if module_name.startswith("numpy"):
-        return "integer" in mro
+    if numpy:
+        loaded_integer_types.extend(
+            [numpy.integer] + get_subclasses(numpy.integer)
+        )
 
-    return isinstance(value, int)
+    return value in loaded_integer_types if type(value) == type else isinstance(value, int)
 
 
 def is_float(value) -> bool:
-    if type(value) != type:
-        value = value.__class__
-
-    mro = get_mro_names(value)
+    if numpy and isinstance(value, numpy.floating):
+        return True
 
-    module_name = value.__module__ if hasattr(value, "__module__") else ''
+    loaded_float_types = [float] + get_subclasses(float)
 
-    if module_name.startswith("numpy"):
-        return "floating" in mro
+    if numpy:
+        loaded_float_types.extend(
+            [numpy.floating] + get_subclasses(numpy.floating)
+        )
 
-    return isinstance(value, float)
+    return value in loaded_float_types if type(value) == type else isinstance(value, float)
```

Comment:
Like the `is_integer` function, I think this can be reduced to 
`return isinstance(value, float) if numpy is None else isinstance(value, (int, np.floating))`

Reply:
Same deal - if we're to cover `is_float(float)`, then we'll need the additional `loaded_float_types` stuff.